packages/flutter/test/material/input_decorator_test.dart

```diff
@@ -364,6 +364,340 @@ void main() {
     expect(tester.getBottomLeft(find.text('label')).dy, tester.getBottomLeft(find.text('hint')).dy);
   });
 
+  testWidgets('InputDecorator input/label widget layout', (WidgetTester tester) async {
+    const Key key = Key('l');
+
+    // The label appears above the input text
+    await tester.pumpWidget(
+      buildInputDecorator(
+        // isEmpty: false (default)
+        // isFocused: false (default)
+        decoration: const InputDecoration(
+          label: Text.rich(
+            TextSpan(
+              children: <InlineSpan>[
+                TextSpan(text: 'label'),
+                WidgetSpan(
+                  child: Text('*', style: TextStyle(color: Colors.red),
+                  ),
+                ),
+              ],
+            ),
+            key: key,
+          ),
+        ),
+      ),
+    );
+    await tester.pumpAndSettle();
+
+    // Overall height for this InputDecorator is 56dps:
+    //   12 - top padding
+    //   12 - floating label (ahem font size 16dps * 0.75 = 12)
+    //    4 - floating label / input text gap
+    //   16 - input text (ahem font size 16dps)
+    //   12 - bottom padding
+
+    expect(tester.getSize(find.byType(InputDecorator)), const Size(800.0, 56.0));
+    expect(tester.getTopLeft(find.text('text')).dy, 28.0);
+    expect(tester.getBottomLeft(find.text('text')).dy, 44.0);
+    expect(tester.getTopLeft(find.byKey(key)).dy, 12.0);
+    expect(tester.getBottomLeft(find.byKey(key)).dy, 24.0);
+    expect(getBorderBottom(tester), 56.0);
+    expect(getBorderWeight(tester), 1.0);
+
+    // The label appears within the input when there is no text content
+    await tester.pumpWidget(
+      buildInputDecorator(
+        isEmpty: true,
+        // isFocused: false (default)
+        decoration: const InputDecoration(
+          label: Text.rich(
+            TextSpan(
+              children: <InlineSpan>[
+                TextSpan(text: 'label'),
+                WidgetSpan(
+                  child: Text('*', style: TextStyle(color: Colors.red),
+                  ),
+                ),
+              ],
+            ),
+            key: key,
+          ),
+        ),
+      ),
+    );
+    await tester.pumpAndSettle();
+
+    expect(tester.getTopLeft(find.byKey(key)).dy, 20.0);
+
+    // The label appears above the input text when there is no content and floatingLabelBehavior is always
```

Comment:
I think you forgot to finish this sentence, unless I'm reading it wrong.

Reply:
This sentence is trying to say that the floatingLabelBehavior is set to always.

Reply:
I'll make this more clear.